src/scene/forward-renderer.js

```diff
@@ -2573,18 +2633,20 @@ Object.assign(pc, function() {
                 this.fogColor[0] = scene.fogColor.r;
                 this.fogColor[1] = scene.fogColor.g;
                 this.fogColor[2] = scene.fogColor.b;
+
                 if (scene.gammaCorrection) {
                     for (i = 0; i < 3; i++) {
                         this.fogColor[i] = Math.pow(this.fogColor[i], 2.2);
                     }
                 }
+
+                this.fogParams[0] = scene.fogDensity / Math.sqrt( Math.log( 2.0 ) );
+                this.fogParams[1] = scene.fogDensity / Math.log( 2.0 );

```

Comment:
Not sure if it makes sense, Math calls can be hardcoded as constants here to improve performance.

Reply:
Agree, will do.

Reply:
Actually - no. JS intepreters are smart enough to pre-compute constants.